core/app/models/spree/country.rb

```diff
@@ -13,9 +13,13 @@ class Country < Spree::Base
 
     validates :name, :iso_name, presence: true
 
+    before_save :clear_cache
+
     def self.default
-      country_id = Spree::Config[:default_country_id]
-      country_id.present? ? find(country_id) : find_by!(iso: 'US')
+      Rails.cache.fetch("#{Rails.application.class.parent_name.underscore}_default_country") do
+        country_id = Spree::Config[:default_country_id]
```

Comment:
As the cached item depends on `Spree::Config[:default_country_id]` is there also a cache invalidation being triggered when this config value is written. Else the readers from the cache could see inconsistent state.